instrumentation/mongo/lib/opentelemetry/instrumentation/mongo/instrumentation.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Mongo
+      # Instrumentation class that detects and installs the Mongo instrumentation
+      class Instrumentation < OpenTelemetry::Instrumentation::Base
+        MINIMUM_VERSION = Gem::Version.new('2.5.0')
+
+        install do |_config|
+          require_dependencies
+          register_tracer_middleware
+        end
+
+        present do
+          !defined?(::Mongo::Monitoring::Global).nil? && gem_version >= MINIMUM_VERSION
+        end
+
+        private
+
+        def gem_version
+          Gem.loaded_specs['mongo']&.version
+        end
+
+        def require_dependencies
+          require_relative 'middlewares/subscriber'
+        end
+
+        def register_tracer_middleware
```

Comment:
It might be more accurate to call this `register_subscriber` or something along those lines.

Reply:
It's also more accurate to call the `Middlewares` module something like `Subscribers`, or perhaps remove that module altogether and move `Subscriber` and friend to the `Mongo` module.